component/src/main/java/io/siddhi/extension/script/js/EvalFunctionExtension.java

```diff
@@ -0,0 +1,164 @@
+        parameters = {
+                @Parameter(name = "expression",
+                        description = "The string with an logical or arithmetic expression.",
+                        type = {DataType.STRING},
+                        dynamic = true),
+                @Parameter(name = "data.type",
+                        description = "The return type of the evaluated expression." +
+                                " Supported types are int/long/float/double/bool/string.",
+                        type = {DataType.STRING},
+                        dynamic = true),
+        },
+        parameterOverloads = {
+                @ParameterOverload(parameterNames = {"expression", "data.type"})
+        },
+        returnAttributes =
+        @ReturnAttribute(
+                description = "The output of the evaluated expression.",
+                type = {DataType.INT, DataType.LONG, DataType.DOUBLE, DataType.FLOAT,
+                        DataType.STRING, DataType.BOOL, DataType.OBJECT}),
+        examples =
+        @Example(
+                syntax = "script:eval(\"700 > 800\", 'bool')",
+                description = "" +
+                        "In this example, the expression 700 > 800 will be evaluated and " +
+                        "return result as false because user specified return type as bool.")
+)
+public class EvalFunctionExtension extends FunctionExecutor {
+
+    private static final String ENGINE_NAME = "nashorn";
+    private ScriptEngine engine;
+    private Attribute.Type returnType = Attribute.Type.OBJECT;
+
+    @Override
+    protected StateFactory<State> init(ExpressionExecutor[] expressionExecutors,
+                                       ConfigReader configReader,
+                                       SiddhiQueryContext siddhiQueryContext) {
+        int executorsCount = expressionExecutors.length;
+
+        if (executorsCount != 2) {
+            throw new SiddhiAppValidationException("Invalid number of arguments passed to "
+                    + "script:eval() function. Required exactly 2, but found " + executorsCount);
+        }
+        ExpressionExecutor executor1 = expressionExecutors[0];
+        ExpressionExecutor executor2 = expressionExecutors[1];
+
+        if (executor1.getReturnType() != STRING) {
+            throw new SiddhiAppValidationException("Invalid parameter type found for the first argument of "
+                    + "script:eval()() function, required " + STRING.toString() + ", but found "
+                    + executor1.getReturnType().toString());
+
+        }
+        if (!(executor2 instanceof ConstantExpressionExecutor)) {
+            throw new SiddhiAppValidationException("The second argument has to be a string constant specifying " +
+                    "one of the supported data types "
+                    + "(int, long, float, double, string, bool)");
+        } else {
+            String type = attributeExpressionExecutors[1].execute(null).toString();
```

Comment:
Why are you passing null to execute(). Cast it to ConstantExpressionExecutor and get the value.
`(String) ((ConstantExpressionExecutor) attributeExpressionExecutors[1]).getValue();`